[html] unresolved import error in embedded JavaScript

Issue Type: <b>Bug</b>

Usually when editing JavaScript files, one can get useful support from TypeScript.
However, when editing JavaScript codes in HTML files, this does not work.

Repro steps:

1. Launch VSCode like `code --disable-extensions`.
2. `File: New Untitled File` (<kbd>⌘ N</kbd>) and set to HTML mode (<kbd>⌘ K</kbd> → <kbd>M</kbd>).
3. Insert the following text:

```html
<script>
  // @ts-check

  Number.parseInt(42);

  /** @typedef {import('lodash')} Lodash */
  /** @type {Lodash['lowerCase']} */
  const lowerCase = () => 42;
</script>
```

Result:

The following problems are detected.

1. Argument of type 'number' is not assignable to parameter of type 'string'. javascript [Ln 4, Col 19]
2. **Cannot find module 'lodash'. Did you mean to set the 'moduleResolution' option to 'node', or to add aliases to the 'paths' option? javascript** [Ln 6, Col 24]

Compare this with (full) JavaScript mode result:

1. Argument of type 'number' is not assignable to parameter of type 'string'. ts(2345) [Ln 3, Col 17]
2. **Type 'number' is not assignable to type 'string'. ts(2322)** [Ln 7, Col 25]

<img width="1392" alt="image" src="https://user-images.githubusercontent.com/4177010/177024428-0b7d0512-17dd-4d82-bee2-18125590b148.png">



VS Code version: Code 1.68.1 (30d9c6cd9483b2cc586687151bcbcd635f373630, 2022-06-14T12:52:13.188Z)
OS version: Darwin x64 20.5.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz (16 x 3600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 2, 3|
|Memory (System)|16.00GB (0.46GB free)|
|Process Argv|--disable-extensions --crash-reporter-id 1c0d2c58-82bd-4296-8754-1c1417025bfb|
|Screen Reader|no|
|VM|0%|
</details>Extensions disabled<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vscscmwlcmt:30465135
cppdebug:30492333
vsclangdc:30486549

```

</details>

<!-- generated by issue reporter -->

--- a/extensions/html-language-features/server/src/modes/javascriptMode.ts
+++ b/extensions/html-language-features/server/src/modes/javascriptMode.ts
@@ -93,13 +93,16 @@ function getLanguageServiceHost(scriptKind: ts.ScriptKind) {
 	};
 }
 
+const ignoredErrors = [
+	1108,  /* A_return_statement_can_only_be_used_within_a_function_body_1108 */
+	2792, /* Cannot_find_module_0_Did_you_mean_to_set_the_moduleResolution_option_to_node_or_to_add_aliases_to_the_paths_option */
+];
 
 export function getJavaScriptMode(documentRegions: LanguageModelCache<HTMLDocumentRegions>, languageId: 'javascript' | 'typescript', workspace: Workspace): LanguageMode {
 	const jsDocuments = getLanguageModelCache<TextDocument>(10, 60, document => documentRegions.get(document).getEmbeddedDocument(languageId));
 
 	const host = getLanguageServiceHost(languageId === 'javascript' ? ts.ScriptKind.JS : ts.ScriptKind.TS);
 	const globalSettings: Settings = {};
-
 	return {
 		getId() {
 			return languageId;
@@ -110,7 +113,7 @@ export function getJavaScriptMode(documentRegions: LanguageModelCache<HTMLDocume
 			const languageService = await host.getLanguageService(jsDocument);
 			const syntaxDiagnostics: ts.Diagnostic[] = languageService.getSyntacticDiagnostics(jsDocument.uri);
 			const semanticDiagnostics = languageService.getSemanticDiagnostics(jsDocument.uri);
-			return syntaxDiagnostics.concat(semanticDiagnostics).filter(d => d.code !== 1108).map((diag: ts.Diagnostic): Diagnostic => {
+			return syntaxDiagnostics.concat(semanticDiagnostics).filter(d => !ignoredErrors.includes(d.code)).map((diag: ts.Diagnostic): Diagnostic => {
 				return {
 					range: convertRange(jsDocument, diag),
 					severity: DiagnosticSeverity.Error,

